cmd/worker_pool.go

```diff
@@ -0,0 +1,74 @@
+package cmd
+
+import bosherr "github.com/cloudfoundry/bosh-utils/errors"
+
+type WorkerPool struct {
+	WorkerCount int
+}
+
+// Runs the given set of tasks in parallel using the configured number of worker go routines
+// Will stop adding new tasks if a task throws an error, but will wait for in-flight tasks to finish
+func (w WorkerPool) ParallelDo(tasks ...func() (interface{}, error)) ([]interface{}, error) {
```

Comment:
@luan thanks for writing up that code snippet and explanation. The aggregation of results and errors still feels like the easier to use worker pool for the CLI, and I don't think adding the `workpool` dep simplifies the code much, might let you remove [this 10 line method](https://github.com/ljfranklin/bosh-cli/blob/3dd3ad58145f1852219bf8499339c120fa8a55f1/cmd/worker_pool.go#L60-L74). I'd vote for leaving the `worker_pool` as is unless @cppforlife objects.